src/pixiewps.c

```diff
@@ -462,10 +474,12 @@ int main(int argc, char **argv) {
 
 	uint_fast8_t k = 0;
 	uint_fast8_t found_p_mode = NONE;
-	unsigned int pin;
+	char *pin;
 	uint32_t seed;
 	uint32_t print_seed = 0;
 
+	pin = calloc(WPS_PIN_LEN + 1, 1); if (!pin) goto memory_err;
```

Comment:
With the change proposed above this line would go.

In any case, to be more precise the line should be (although in C99 I believe sizeof(char) is always 1):
```C
pin = calloc(WPS_PIN_LEN + 1, sizeof(char)); if (!pin) goto memory_err;
```